src/tiled/command.cpp

```diff
@@ -30,39 +30,70 @@
 using namespace Tiled;
 using namespace Tiled::Internal;
 
+QString Command::finalWorkingDirectory() const
+{
+    QString finalWorkingDirectory = workingDirectory;
+
+    finalWorkingDirectory = replaceVariables(finalWorkingDirectory, false);
+
+    QString finalExecutable = replaceVariables(executable);
+    QFileInfo mFile(finalExecutable);   //Check if executable is a path or not
+
+    if (mFile.exists() && mFile.isFile()) {
+        finalWorkingDirectory.replace(QLatin1String("%executablepath"),
+                            mFile.absolutePath());
+    } else {
+        finalWorkingDirectory.replace(QLatin1String("%executablepath"),
+                            QString(QLatin1String("")));
```

Comment:
In case you leave this here, note that to create an empty string you should just write `QString()`, and to remove something from a string, you would use `finalWorkingDirectory.remove(...)`.